.github/workflows/doc.yml

```diff
@@ -27,12 +27,69 @@ jobs:
       CACHE_GALLERY_EXAMPLES: './doc/examples'
       CACHE_POOCH: 'pyxem'
 
-  Push_on_tag:
+  Deploy_dev_docs:
     needs: Build
-    # Push only on tag and update the "current" documentation
-    if: ${{ github.repository_owner == 'pyxem' && startsWith(github.ref, 'refs/tags/') }}
-    permissions:
-      # needs write permission to push the docs to gh-pages
-      contents: write
-    # Use the "reusable workflow" from the hyperspy organisation
-    uses: hyperspy/.github/.github/workflows/push_doc.yml@main
+    runs-on: ubuntu-latest
+    steps:
+      # Deploy docs for main branch -> /dev/
+      - name: Deploy dev docs
+        if: github.ref == 'refs/heads/main'
+        uses: peaceiris/actions-gh-pages@v3
```

Comment:
Since this is a third party action, should it be pinned to the full commit SHA, as per guidance in https://docs.github.com/en/actions/reference/security/secure-use#using-third-party-actions? This will be updated by dependabot.

Reply:
Good point. I was a little worried about that as a possible security threat.